Docs

Akrog opened the pull request:
This PR includes a series of patches to improve the cinder operator documentation to include:

- Getting started guide
- Contributor guidelines
- Using external dependencies
- Debugging guide

Akrog committed:
Add hacking files

This patch adds a series of files to the `hack` directory to facilitate
the development.

These files include:

- Helper functions
- Script to deploy a local toy Ceph cluster
- Script to create the LVM loopback device inside the OpenShift VM
- OpenStack and Cinder client installation and configuration files
- Manifest for OpenStack with Cinder and Glance using local Ceph cluster

Akrog committed:
Docs: Update README.md

This patch updates the README.md file to be a bit more detailed in how
we can get a working deployment with Ceph as a backend for Cinder and
Glance.

Akrog committed:
Docs: Contributor guide

This patch adds some documentations to help new contributors get
started.

It is not complete, since it doesn't explain how to work with merged and
unmerged dependencies with other operators and lib-common, but it should
serve as a good start.

Akrog committed:
Docs: External dependencies

This patch introduces a couple of guides on how to do dependencies in
the cinder-operator to explain simple and circular references for both
running the operator locally and building the operator.

Akrog committed:
Docs: Add debug guide

This patch introduces a simple debug guide to help new developers.

Akrog commented:
I need to update these commits for the new CRD from https://github.com/openstack-k8s-operators/openstack-operator/pull/53

dprince commented on hack/dev/openstack-ceph.yaml:
```diff
@@ -0,0 +1,114 @@
+apiVersion: core.openstack.org/v1beta1
```
only negative about having an example like this within the cinder-operator is it is likely to become stale quickly. Would this specific example be better suited within the openstack-operator itself?

Akrog replied:
My idea of having this example here for now was to promote cinder folks *always* deploy using the openstack-operator and for now to make sure we *always* keep it updated.

In the near future (once we merge the `extraVolumes` stuff) I'm think of adding it like you suggest to the openstack-operator itself and make the `openstack_deploy` target in `install_yamls`' Makefile to do the ceph _toy_ cluster deployment automatically and use it in Cinder and Glance, that way we get something similar to what devstack does, a working OpenStack deployment with Ceph.

fultonj replied:
Maybe we can merge PR77 with `openstack-ceph.yaml` and then remove it once a copy of it is in the openstack-operator.

Akrog commented:
> 

You read my mind.  :-)
First get something here and then see what things are common and can be moved to the docs repo.

Akrog commented:
Do not merge, I want to update it to use workspaces once we merge https://github.com/openstack-k8s-operators/cinder-operator/pull/79

Akrog commented:
/hold

Akrog committed:
Hack: Testing PR helpers

This patch adds some scripts to facilitate the testing of PRs with
dependencies.

Four scripts are introduced:

- hack/cleandeps.sh: Cleans up contents of go.work

- hack/showdeps.py: Show the dependencies of a PR based on its commit
  messages contents (Depends-On:)

- hack/setdeps.py: Replaces modules in the go workspace with local
  directories, PRs, remote branches, etc.

- hack/checkout_pr.sh: Leverages the other scripts to do the checkout of
  a PR and replace the dependencies as needed.

Akrog committed:
Docs & Hack: extraVol changes

This patch makes changes to the getting started scripts and manifest to
adapt to the new extraVol changes.

tosky commented on docs/dev/local-dependencies.md:
````diff
@@ -0,0 +1,286 @@
+Alternatively we can use the `hack/setdeps.py` to do all that for us:
+
+```
+hack/setdeps.py lib-common=88
````
I probably don't know the syntax, but is this really 88?

tosky replied:
looking at he next like it seems 88 is the pull request ID, could it be explicitly mention?

fultonj commented on README.md:
````diff
@@ -1,149 +1,242 @@
-### How it works
-This project aims to follow the Kubernetes [Operator pattern](https://kubernetes.io/docs/concepts/extend-kubernetes/operator/)
+```sh
+source ~/cinder-operator/hack/dev/osp-clients-cfg.sh
````
This works but maybe in the future it can be released by the client pod.

fultonj commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,241 @@
+
+The cinder-operator project will not squash all pull request commits into a
+single commit but will look to preserve all submitted individual commits
+instead using a merge strategy instead.  This means that we can have both
```
Please remove one of the 'instead'.

fultonj commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,241 @@
+
+#### PR Images
+
+Once a PR merges it will trigger an image rebuild and publish, so how can we
```
Good question. I don't know but I'm commenting here to bring attention to other reviewers who might know. @dprince do you know?

fultonj reviewed (commented):
This is really helpful. Thank you Gorka. I think these docs help a lot and aside from a few typos it would be nice to see them merged in the project.

fultonj commented on docs/dev/local.md:
```diff
@@ -0,0 +1,135 @@
+only have a cinder-operator pod running, but also the different cinder
+services.
+
+Since we have everything running we need to uninstalling both the
```
s/uninstalling/uninstall
